packages/elastic-safer-lodash-set/scripts/update.sh

```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env bash
+
+# Elasticsearch B.V licenses this file to you under the MIT License.
+# See `packages/elastic-safer-lodash-set/LICENSE` for more information.
+
+set -e
+
+source ./scripts/_get_lodash.sh
+
+all_files=$(cd lodash && ls)
+modified_lodash_files=(_baseSet.js)
+
+# Get fresh copies of all the files that was originally copied from lodash,
+# expect the ones in the whitelist
+for file in $all_files
+do
+  if [[ ! "${modified_lodash_files[@]}" =~ "${file}" ]]
```

Comment:
__note:__ doesn't matter right now since we have just one file, but it can become problematic in the future see https://github.com/koalaman/shellcheck/wiki/SC2199

Reply:
If I understand it correctly, it should still work even though it's concatenated to a string containing all the elements. It's essentially just checking that the string on the right (`file`) is present somewhere in the string on the right (the space-concatenated version of the `modified_lodash_files` array). Or maybe I'm misunderstanding the problem?

Reply:
Yeah, it works like that just fine. It was just pedantic super minor nit, feel free to ignore. I meant that we don't need quotes in the concatenated string:
```bash
modified_lodash_files=(_baseSet.js _set.js)

printf "'%s'\n" "${modified_lodash_files[*]}"
'_baseSet.js _set.js'

printf "'%s'\n" "${modified_lodash_files[@]}"
'_baseSet.js''_set.js'
```

Reply:
Not sure I fully understand what you mean 😓 But as you say, there's no need to let this be a showstopper for merging. With your help, I'd love to improve it afterwards 🙂 

I played around with some arrays to try all the edge cases I could think off. To support multi-word words I had to add quotes around `${all_files[@]}`. Here's my test code:

```bash
all_files=(foobar.js bar.js "multi word")
modified_lodash_files=(foobar.js "foo bar")

echo "ALL FILES:"
echo "---------------------------------------------"
for file in "${all_files[@]}"
do
  echo "  ${file}"
done

echo
echo "MODIFIED FILES:"
echo "---------------------------------------------"
for file in "${modified_lodash_files[@]}"
do
  echo "  ${file}"
done

echo
echo "Looping over all files:"
echo "---------------------------------------------"
for file in "${all_files[@]}"
do
  echo "Checking '${file}':"
  if [[ ! "${modified_lodash_files[@]}" =~ "${file}" ]]
  then
    echo "  NOT INCLUDED IN MODIFIED LIST: ${file}"
  else
    echo "  INCLUDED IN MODIFIED LIST: ${file}"
  fi
done
```

Here's the output:

```
ALL FILES:
---------------------------------------------
  foobar.js
  bar.js
  multi word

MODIFIED FILES:
---------------------------------------------
  foobar.js
  foo bar

Looping over all files:
---------------------------------------------
Checking 'foobar.js':
  INCLUDED IN MODIFIED LIST: foobar.js
Checking 'bar.js':
  INCLUDED IN MODIFIED LIST: bar.js
Checking 'multi word':
  NOT INCLUDED IN MODIFIED LIST: multi word
```

As you can see, the main issue is that it will match inside of filenames as well. But as far as I can see, that's it.

Reply:
Oh I just realized that there's a difference between `arr=(1 2 3)` and `arr=$(ls)`... With the latter I can't loop over the array with `"${arr[@]}"`, but have to do it without the quotes: `${arr[@]}`. Downside of that is that it doesn't support filenames that include whitespace 🤔

Reply:
> With your help, I'd love to improve it afterwards slightly_smiling_face

Sounds good :+1: